### PR TITLE
Document and improve translation process

### LINKED ISSUE #4516: Add documentation for translators
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

We've had some recent volunteers for translations to new languages, and we've had a lot of changes to translatable strings.  We feel there could be more documentation and support for translators and developers working with these strings.

### Describe the possible solution

We'd like to add some documentation for future translators and developers to make the process smoother.

### Additional info

See PR #4452 for more info on the process.

### PR DESCRIPTION
**Pull request by yvesgurcan**
Closes #4516. 

- Adds documentation on how to work on translations.
- Updates translation `Makefile` to enable copying MO files to build folders and clean up after creating MO files.
- Updates README with minor improvements.
- Adds `.dat` files to the list of ignored files (these files can be found in the `/data` folder and shouldn't be committed).

### CODE REVIEW

**Comment by zenseii** [comment]
Would you be able to make a preview of your .md changes similar to the one I made here for the translation.md mock-up:
https://github.com/zenseii/fheroes2/wiki/Translation-Guide ? 

Also, the translation guide should keep in mind that it should be possible to understand for people with zero background in programming, which is why an explanation of terminology is necessary.

File: files/lang/Makefile [hunk 1]
```diff
@@ -2,13 +2,23 @@
...
 
+copy:
+	cp *.mo ../../build/x64/Debug-SDL1/files/lang 2>/dev/null || :
+	cp *.mo ../../build/x86/Debug-SDL1/files/lang 2>/dev/null || :
+	cp *.mo ../../build/x64/Debug-SDL2/files/lang 2>/dev/null || :
+	cp *.mo ../../build/x86/Debug-SDL2/files/lang 2>/dev/null || :
+	cp *.mo ../../build/x64/Release-SDL1/files/lang 2>/dev/null || :
+	cp *.mo ../../build/x86/Release-SDL1/files/lang 2>/dev/null || :
+	cp *.mo ../../build/x64/Release-SDL2/files/lang 2>/dev/null || :
+	cp *.mo ../../build/x86/Release-SDL2/files/lang 2>/dev/null || :
```
Comment (oleg-derevenetz) [inline files/lang/Makefile:20]: How exactly are these build directories created and why are they exactly like this and not some others? In general, binary should either be launched from the root of the source tree or `FHEROES2_DATA` environment variable should be set if for some reason it is launched from a non-standard location. There is also the `CONFIGURE_FHEROES2_DATA` preprocessor definition to set the location of `files/data` and `files/lang` directories at compile time. This is not a matter of the location of the translation files alone - for example, `resurrection.h2d` will not be found too, and `data` and `maps` directories from the source tree will not be found as well without this.

File: files/lang/Makefile [hunk 2]
```diff
@@ -18,18 +28,21 @@ es.mo: es.po
 	sed -e 'y/äëïöőüűÄËÏŐÖÜŰ/aeioouuAEIOOUU/' $< > $<.tmp
 	LANG=es_ES.UTF-8 LC_ALL=es_ES.UTF-8 LC_CTYPE=es_ES.UTF-8 iconv -f utf-8 -t ascii//TRANSLIT $<.tmp > $<.2.tmp
 	msgfmt $<.2.tmp -o $@
+	rm -f $<.tmp $<.2.tmp
```
Comment (oleg-derevenetz) [inline files/lang/Makefile:31]: All these `*.tmp` files will be removed at the `make clean` stage, is there any real reason to delete them immediately?

**Review by oleg-derevenetz — CHANGES_REQUESTED** [review]
Hi @yvesgurcan I just left a few notices and questions regarding this pull request, please take a look when it's convenient for you.

File: docs/TRANSLATION.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# [**fheroes2**](README.md) translation guide
```
Comment (ihhub) [inline docs/TRANSLATION.md:1]: We need to add a note that this documentation is only for Linux / MacOS users. An ordinary Windows user can't do such commands.

File: docs/TRANSLATION.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# [**fheroes2**](README.md) translation guide
+
+This project uses portable object (PO) files to handle localization in various languages.
+
+## Finding translatable strings in the codebase
+
+Translatable strings can be found in the code source as arguments to the `_` function (which is a short name for `gettext`). The string below, for example, can be translated in PO files:
```
Comment (ihhub) [inline docs/TRANSLATION.md:7]: We also have `_n` function which is for plural forms.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @yvesgurcan , I left few comments as well. Could you please take a look?

File: docs/TRANSLATION.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+Once the POT file has been created, go to `/files/lang` and run the following command to update translatable strings in the PO files:
+
+```bash
+make *.po clean
+```
+
+## Editing translations
+
+The PO files are located in `/files/lang`. We encourage you to use [*poedit**](https://poedit.net/) to edit translations.
```
[the remark this answers is not in the document]
Comment (dimag0g) [reply docs/TRANSLATION.md:29]: I suggest we at least mention [gtranslator](https://wiki.gnome.org/Apps/Gtranslator) for those who would like to use completely open-source software. Poedit has an open-source version too (https://github.com/vslavik/poedit), but the version from https://poedit.net/ is not.

**Comment by ihhub** [comment]
Hi @oleg-derevenetz , I am thinking that for now we should remove **copy** stage and revert all changes within Makefile. This will make the documentation smaller and we can proceed with merging the rest of changes. What do you think?

**Comment by oleg-derevenetz** [comment]
Hi @ihhub yes, I think so too. We will also need to remove the mention of `make copy` from the `TRANSLATION.md`.

**Review by ihhub — APPROVED** [review]

File: docs/TRANSLATION.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# [**fheroes2**](README.md) translation guide
+
+This project uses portable object (PO) files to handle localization in various languages. The current instruction is design for Linux/MacOS users. Windows users should install [Cygwin](https://www.cygwin.com/) in order to be able to work with translations.
```
Comment (a1exsh) [inline docs/TRANSLATION.md:3]:
```suggestion
This project uses portable object (PO) files to handle localization in various languages. The current instruction is designed for Linux/MacOS users. Windows users should install [Cygwin](https://www.cygwin.com/) in order to be able to work with translations.

```

File: docs/TRANSLATION.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# [**fheroes2**](README.md) translation guide
+
+This project uses portable object (PO) files to handle localization in various languages. The current instruction is design for Linux/MacOS users. Windows users should install [Cygwin](https://www.cygwin.com/) in order to be able to work with translations.
+
+## Finding translatable strings in the codebase
+
+Translatable strings can be found in the source code as arguments to the `_` or `_n` functions (which are short names for `gettext` and `ngettext` functions respectively). The string below, for example, can be translated in PO files:
```
Comment (a1exsh) [inline docs/TRANSLATION.md:7]: What about `gettext_noop`?

**Comment by ihhub** [comment]
Hi @a1exsh , I addressed your given comments.

**Review by oleg-derevenetz — APPROVED** [review]

**Comment by ihhub** [comment]
@yvesgurcan , thank you so much for your contribution to the project!

**Pull request merged**